### DISCUSSION TITLE
Text to Speech help / documentation

### DISCUSSION THREAD

**Discussion by villares** [johnfactotum/foliate#1200] [Q&A] [answered]
From the FAQ [how-to-use-text-to-speech](https://github.com/johnfactotum/foliate/blob/gtk4/docs/faq.md#how-to-use-text-to-speech) there is this single paragraph:

> Foliate supports text-to-speech with speech-dispatcher. To use it, you need to install speech-dispatcher and output modules such as espeak-ng.

As I installed Foliate it asked for optional TTS dependencies, I installed `festival`, `festival-english` voices, and also `espeak-ng`. Then reading the FAQ I installed `speech-dispatcher`. But there is no hint on Foliate's interface on how to enable the speech feature. I have no idea of what to do next. What am I missing? Could the TTS instructions be expanded a bit maybe?

After another search I found a similar *troubleshooting* paragraph that mentioned testing `speetch-dispatcher` with `spd-say`. It worked for me (`spd-say "hello"` said "hello" ).

**Comment by johnfactotum** [comment]
There's nothing to enable. When you click on the play button it will try to spawn `speech-dispatcher` and send messages to it.

If you're using Flatpak or Snap, or other sandboxed environments, it won't be able to spawn `speech-dispatcher` and you would either need to enable socket activation (it seems currently only Ubuntu ships `speech-dispatcher` with socket activation enabled) or keep speech-dispatcher running manually yourself.

**Reply by pangyuehung** [reply]
> enable socket activation

how?

**Reply by johnfactotum** [reply]
It's a compile time option in `speech-dispatcher`, if I remember correctly. It's something that should be configured by the distribution.

**Comment by villares** [comment]
Cheers @johnfactotum, thanks for Foliate and for reaching out!

Where is the *play button* supposed to appear on the interface? I can't find it.

I have installed Foliate with Manjaro's *official repositories (extra)* feature, it is not Flatpak nor snap.

**Reply by johnfactotum** [reply, marked as answer]
There's a <i>Narration</i> button (the one with a headphones icon) on the navbar (which is available by hovering or tapping on the footer area).
 
Note that if the book has embedded audio (media overlays), the <i>Narration</i> button would show controls for the embedded media, and TTS would not be available in that case.

Alternatively you can select some text and choose <i>Speak from here</i> from the selection menu. Though you still need to use the <i>Narration</i> button if you want to stop the speech output.

**Reply by villares** [reply]
Ok! So now I found the headphones icon, thank you!

Could [this answer](https://github.com/johnfactotum/foliate/blob/gtk4/docs/faq.md#how-to-use-text-to-speech) be expanded a bit  with your previous info?

**Reply by johnfactotum** [reply]
Done: 1c55b0a4a525edd3a38c126875fd60e7f732439c.
